src/OpenTelemetry.Exporter.Console/ConsoleLogRecordExporter.cs

```diff
@@ -129,5 +140,37 @@ void ProcessScope(LogRecordScope scope, ConsoleLogRecordExporter exporter)
 
             return ExportResult.Success;
         }
+
+        protected override void Dispose(bool disposing)
+        {
+            lock (this.disposeLockObj)
+            {
+                if (!this.disposed)
+                {
+                    this.disposed = true;
+                    this.disposedStackTrace = Environment.StackTrace;
+                }
+            }
+
+            base.Dispose(disposing);
+        }
+
+        private void SendDisposedMessage()
+        {
+            lock (this.disposeLockObj)
```

Comment:
a double-lock pattern seems to make more sense here

Reply:
Thanks @reyang , updated double-checking locking and changelog, can this be resolved?

Reply:
Looks better! The method name `SendDisposedMessage` is now a bit misleading, the name suggested that the method will send the message, but the actual method is doing something different (checking to see if it's the first time). I would suggest to get rid of this method, and put the logic directly in `Export` (an alternative approach would be keeping the 4 print lines in the method, and put the double locking logic in `Export`, but I feel that 4 print lines won't justify having its own method).

Reply:
Updated